pkg/fs/s3.go

```diff
@@ -24,13 +24,24 @@ type S3Config struct {
 	Region string `toml:"region"`
 	// EndpointURL is an HTTP endpoint of the S3 API
 	EndpointURL string `toml:"endpoint_url"`
+	// Prefix is a prefix (subfolder) to use to build key names
+	Prefix string `toml:"prefix"`
 }
 
 // S3 implements file storage for S3-compatible providers.
 type S3 struct {
 	api      s3iface.S3API
 	uploader *s3manager.Uploader
 	bucket   string
+	prefix   string
+}
+
+func buildKey(name string, s *S3) string {
+	if len(s.prefix) == 0 {
+		return name
+	}
+	result := s.prefix + "/" + name
+	return result
 }
```

Comment:
You can simply use `path.Join(s.prefix, name)` instead of implementing a helper function.

Reply:
thanks, I used t in a helper function.
But I think helper is still nice to have to simplify code and add some testing